expr/bitwise_test.go

```diff
@@ -0,0 +1,55 @@
+package expr
+
+import (
+	"encoding/binary"
+	"reflect"
+	"testing"
+
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+)
+
+func TestBitwise(t *testing.T) {
+	tests := []struct {
+		name string
+		bw   Bitwise
+	}{
+		{
+			name: "Unmarshal Bitwise IPv4 case",
+			bw: Bitwise{
+				SourceRegister: 1,
+				DestRegister:   2,
+				Len:            4,
+				Xor:            []byte{0x0, 0x0, 0x0, 0x0},
+				Mask:           []byte{0xff, 0xff, 0x0, 0x0},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		nbw := Bitwise{}
+		data, err := tt.bw.marshal()
+		if err != nil {
+			t.Errorf("Test \"%s\" failed to marshal Bitwise struct with error: %+v", tt.name, err)
+			continue
```

Comment:
Instead of t.Errorf+continue and prefixing each error with the tests name, please use subtests with t.Fatalf: http://blog.golang.org/subtests

Reply:
done

Reply:
This does not seem to be addressed yet? I don’t see you using a subtest even in the latest changes?

Reply:
Please check diff again. it has all been done. Not sure why you do not see it.

Reply:
I don’t see a subtest (t.Run), even after clearing my browser cache. Looking into your branch, I don’t see it either: https://github.com/sbezverk/nftables/blob/add_bitwise_decoder/expr/bitwise_test.go

Did you forget to push a commit, or misunderstand what I meant?

Reply:
https://github.com/sbezverk/nftables/blob/add_bitwise_decoder/expr/bitwise_test.go#L36

```
	for _, tt := range tests {
		nbw := Bitwise{}
		data, err := tt.bw.marshal()
		if err != nil {
			t.Fatalf("Test \"%s\" failed to marshal Bitwise struct with error: %+v", tt.name, err)

		}

```

I tried force push, but it says no changes. Not sure what else can be done..

Reply:
In your code excerpt, I don’t see t.Run either?

Reply:
Right, I thought your comment was just about Errorf/continue vs Fatalf.